fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/InternalToCertificate.java

```diff
@@ -681,4 +772,866 @@ private static CertificateDataValueDiagnosis createDiagnosis(String id, Diagnos
+                    .list(
+                        Arrays.asList(
+                            CheckboxDateRange.builder()
+                                .id(SjukskrivningsGrad.NEDSATT_1_4.getId())
+                                .label(SjukskrivningsGrad.NEDSATT_1_4.getLabel())
+                                .build(),
+                            CheckboxDateRange.builder()
+                                .id(SjukskrivningsGrad.NEDSATT_HALFTEN.getId())
+                                .label(SjukskrivningsGrad.NEDSATT_HALFTEN.getLabel())
+                                .build(),
+                            CheckboxDateRange.builder()
+                                .id(SjukskrivningsGrad.NEDSATT_3_4.getId())
+                                .label(SjukskrivningsGrad.NEDSATT_3_4.getLabel())
+                                .build(),
+                            CheckboxDateRange.builder()
+                                .id(SjukskrivningsGrad.HELT_NEDSATT.getId())
+                                .label(SjukskrivningsGrad.HELT_NEDSATT.getLabel())
+                                .build()
+                        )
+                    )
+                    .build()
+            )
+            .value(
+                CertificateDataValueDateRangeList.builder()
+                    .list(createSjukskrivningValue(internalCertificate))
+                    .build()
+            )
+            .validation(
+                new CertificateDataValidation[]{
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(BEHOV_AV_SJUKSKRIVNING_SVAR_ID_32)
+                        .expression(multipleOrExpression(
+                            singleExpression(SjukskrivningsGrad.NEDSATT_1_4.getId()),
+                            singleExpression(SjukskrivningsGrad.NEDSATT_HALFTEN.getId()),
+                            singleExpression(SjukskrivningsGrad.NEDSATT_3_4.getId()),
+                            singleExpression(SjukskrivningsGrad.HELT_NEDSATT.getId())
+                        ))
+                        .build()
+                }
+            )
+            .build();
+    }
+
+    private static List<CertificateDataValueDateRange> createSjukskrivningValue(LisjpUtlatandeV1 internalCertificate) {
+        if (internalCertificate == null) {
+            return Collections.emptyList();
+        }
+        return internalCertificate.getSjukskrivningar().stream()
+            .map(item -> CertificateDataValueDateRange.builder()
+                .id(item.getSjukskrivningsgrad().getId())
+                .to(item.getPeriod().getTom().asLocalDate())
+                .from(item.getPeriod().getFrom().asLocalDate())
+                .build()
+            ).collect(Collectors.toList());
+    }
+
+    private static CertificateDataElement createMotiveringTidigtStartdatumQuestion(LisjpUtlatandeV1 internalCertificate, int index) {
+        final var endOfExpression = ".from < -7";
```

Comment:
Should be a different way of implementing this. It is hard to understand what the "endOfExpression" really means. Would be good to expand the expression-toolkit with more logic.